codegen/src/codegen.rs

```diff
@@ -21,9 +21,32 @@ use std::{
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 #[derive(Debug, Snafu)]
 pub enum Error {
-    SpecError { source: spec::Error },
+    SpecError {
+        source: spec::Error,
+    },
     ArrayExpectedToHaveItems,
     NoNameForRef,
+    EmtpyIdentError,
+    #[snafu(display("ParseIdentError {} {}", text, source))]
+    ParseIdentError {
+        source: syn::Error,
+        text: String,
+    },
+    #[snafu(display("IdentError at {}:{} {} ", file, line, source))]
+    IdentError {
+        #[snafu(source(from(Error, Box::new)))]
+        source: Box<Error>,
+        file: &'static str,
```

Comment:
It would be amazing if `file` and `line` could be automatically set like `source`.